app/src/main/java/io/intrepid/contest/screens/contestoverview/ContestOverviewActivity.java

```diff
@@ -45,10 +66,27 @@ public void showContestName(String contestName) {
     }
 
     @Override
+    public void showContestDescription(String description) {
+        descriptionTextView.setText(description);
+    }
+
+    @Override
+    public void showCategories(List<Category> categories) {
+        categoryAdapter.setCategories(categories);
+        categoriesRecyclerView.setAdapter(categoryAdapter);
```

Comment:
Do you need to set the adapter again every time the categories are updated? Is there another way to notify the changes?
(This might be the right way, this is more a question than a recommendation)